types.go

```diff
@@ -74,10 +68,10 @@ type Channel interface {
 
 	// Selector returns the IPLD selector for this data transfer (represented as
 	// an IPLD node)
-	Selector() ipld.Node
+	Selector() datamodel.Node
 
-	// Voucher returns the voucher for this data transfer
-	Voucher() Voucher
+	// Voucher returns the initial voucher for this data transfer
+	Voucher() (TypedVoucher, error)
```

Comment:
I think this is ultimately the correct change and something we should have done a while ago. That said, keep in mind this will bubble up a lot of api changes. We had better factor in integration with Lotus + Filclient + Boost at a minimum in our timelines